[Bug]: close of closed channel in Gemini plugin after Vertex AI 429 error (Resource exhausted)

### What happened?

When using Fabric with the Gemini plugin, the CLI panics with `close of closed channel` after a `RESOURCE_EXHAUSTED` (HTTP 429) error is returned from Google Vertex AI.

It seems that the error handling in the Gemini client triggers a double channel close, causing a runtime panic.

---

### **Steps to Reproduce**

1. Configure Fabric to use the Gemini AI provider.
2. Run a command repeatedly (e.g., in quick succession):

   ```bash
   wl-paste | fabric -sp improve_writing
   ```
3. After several requests, once the Vertex AI quota or rate limit is reached, Fabric throws a 429 error and immediately panics.

---

### **Observed Behavior**

```
Error: Error 429, Message: Resource exhausted. Please try again later.
panic: close of closed channel

goroutine 9 [running]:
github.com/danielmiessler/fabric/internal/plugins/ai/gemini.(*Client).SendStream(...)
    /path/to/fabric/internal/plugins/ai/gemini/gemini.go:165
github.com/danielmiessler/fabric/internal/core.(*Chatter).Send.func1(...)
    /path/to/fabric/internal/core/chatter.go:75
created by github.com/danielmiessler/fabric/internal/core.(*Chatter).Send
    /path/to/fabric/internal/core/chatter.go:73 +0x2fd
```

---

### **Expected Behavior**

Fabric should gracefully handle the Vertex AI 429 error (rate limit or quota exhaustion) and display an informative message to the user without panicking.

---

### **Environment**

* **Fabric version:** `v1.4.321`
* **Go version:** `go1.25.3 linux/amd64`
* **OS:** Linux (Ubuntu 24.04.3 LTS)
* **Backend:** `Gemini / Vertex AI`
* **Command:** `fabric -sp improve_writing`

### Operating System

Linux - amd64

### OS Version

```shell
PRETTY_NAME="Ubuntu 24.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.3 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
```

### How did you install Fabric?

Release Binary - Linux (amd64)

### Version

```text
v1.4.321
```

### Relevant log output

```shell
panic: close of closed channel

goroutine 21 [running]:
github.com/danielmiessler/fabric/internal/plugins/ai/gemini.(*Client).SendStream(0xc000475530, {0xc0001993b8, 0x1, 0x1}, 0xc00040b180, 0xc0001848c0)
        /home/ubuntu/go/pkg/mod/github.com/danielmiessler/fabric@v1.4.321/internal/plugins/ai/gemini/gemini.go:165 +0x292
github.com/danielmiessler/fabric/internal/core.(*Chatter).Send.func1()
        /home/ubuntu/go/pkg/mod/github.com/danielmiessler/fabric@v1.4.321/internal/core/chatter.go:75 +0xc6
created by github.com/danielmiessler/fabric/internal/core.(*Chatter).Send in goroutine 1
        /home/ubuntu/go/pkg/mod/github.com/danielmiessler/fabric@v1.4.321/internal/core/chatter.go:73 +0x2fd
```

### Relevant screenshots (optional)

_No response_

--- a/cmd/generate_changelog/incoming/1832.txt
+++ b/cmd/generate_changelog/incoming/1832.txt
@@ -1,0 +1,7 @@
+### PR [#1832](https://github.com/danielmiessler/Fabric/pull/1832) by [ksylvan](https://github.com/ksylvan): Improve channel management in Gemini provider
+
+- Fix: improve channel management in Gemini streaming method
+- Add deferred channel close at function start
+- Return error immediately instead of breaking loop
+- Remove redundant channel close statements from loop
+- Ensure channel closes on all exit paths consistently

--- a/internal/plugins/ai/gemini/gemini.go
+++ b/internal/plugins/ai/gemini/gemini.go
@@ -131,6 +131,8 @@ func (o *Client) Send(ctx context.Context, msgs []*chat.ChatCompletionMessage, o
 
 func (o *Client) SendStream(msgs []*chat.ChatCompletionMessage, opts *domain.ChatOptions, channel chan string) (err error) {
 	ctx := context.Background()
+	defer close(channel)
+
 	var client *genai.Client
 	if client, err = genai.NewClient(ctx, &genai.ClientConfig{
 		APIKey:  o.ApiKey.Value,
@@ -153,8 +155,7 @@ func (o *Client) SendStream(msgs []*chat.ChatCompletionMessage, opts *domain.Cha
 	for response, err := range stream {
 		if err != nil {
 			channel <- fmt.Sprintf("Error: %v\n", err)
-			close(channel)
-			break
+			return err
 		}
 
 		text := o.extractTextFromResponse(response)
@@ -162,7 +163,6 @@ func (o *Client) SendStream(msgs []*chat.ChatCompletionMessage, opts *domain.Cha
 			channel <- text
 		}
 	}
-	close(channel)
 
 	return
 }

